environment/environment.go

```diff
@@ -805,31 +771,31 @@ func (m *Environment) findPodCountInDeploymentManifest() int {
 			continue
 		}
 		for _, j := range *json {
-			m := j.(map[string]interface{})
+			m := j.(map[string]any)
 			// if the kind is a deployment then we want to see if it has replicas to count towards the app count
 			kind := m["kind"].(string)
 			if kind == "Deployment" || kind == "StatefulSet" {
-				podCount += getReplicaCount(m["spec"].(map[string]interface{}))
+				podCount += getReplicaCount(m["spec"].(map[string]any))
 			}
 		}
 
 	}
 	return podCount
 }
 
-func getReplicaCount(spec map[string]interface{}) int {
+func getReplicaCount(spec map[string]any) int {
 	if spec == nil {
 		return 0
 	}
-	s := spec["selector"].(map[string]interface{})
+	s := spec["selector"].(map[string]any)
```

Comment:
same comment for the check regarding the corresponding key existence in the map. might throw panic

Reply:
Updated and also did checks for the ones below in this function as well